packages/flutter_tools/lib/src/isolated/native_assets/android/native_assets.dart

```diff
@@ -189,36 +186,37 @@ AndroidArch _getAndroidArch(Target target) {
   }
 }
 
-Map<Asset, KernelAsset> _assetTargetLocations(List<Asset> nativeAssets) {
-  return <Asset, KernelAsset>{
-    for (final Asset asset in nativeAssets)
+Map<AssetImpl, KernelAsset> _assetTargetLocations(
+    List<AssetImpl> nativeAssets) {
+  return <AssetImpl, KernelAsset>{
+    for (final AssetImpl asset in nativeAssets)
       asset: _targetLocationAndroid(asset),
   };
 }
 
 /// Converts the `path` of [asset] as output from a `build.dart` invocation to
 /// the path used inside the Flutter app bundle.
-KernelAsset _targetLocationAndroid(Asset asset) {
-  final AssetPath path = asset.path;
+KernelAsset _targetLocationAndroid(AssetImpl asset) {
+  final LinkModeImpl linkMode = (asset as NativeCodeAssetImpl).linkMode;
   final KernelAssetPath kernelAssetPath;
-  switch (path) {
-    case AssetSystemPath _:
-      kernelAssetPath = KernelAssetSystemPath(path.uri);
-    case AssetInExecutable _:
+  switch (linkMode) {
+    case DynamicLoadingSystemImpl _:
+      kernelAssetPath = KernelAssetSystemPath(linkMode.uri);
+    case LookupInExecutableImpl _:
       kernelAssetPath = KernelAssetInExecutable();
-    case AssetInProcess _:
+    case LookupInProcessImpl _:
       kernelAssetPath = KernelAssetInProcess();
-    case AssetAbsolutePath _:
-      final String fileName = path.uri.pathSegments.last;
+    case DynamicLoadingBundledImpl _:
+      final String fileName = asset.file!.pathSegments.last;
       kernelAssetPath = KernelAssetAbsolutePath(Uri(path: fileName));
     default:
       throw Exception(
-        'Unsupported asset path type ${path.runtimeType} in asset $asset',
+        'Unsupported asset link mode $linkMode in asset $asset',
       );
   }
   return KernelAsset(
     id: asset.id,
-    target: asset.target,
+    target: Target.fromArchitectureAndOS(asset.architecture!, asset.os),
```

Comment:
why was this change necessary?

Reply:
`Target` is removed from the `hook/build.dart` API using `Asset`s, but still used in `KernelAsset`. `Target` is a VM concept, `hook/build.dart` writers will only talk about `OS` and `Architecture`. `dartdev` and `flutter_tools` have to do the conversion.

Reply:
Thanks